server/src/main/scala/com/cloudera/livy/server/LivyServer.scala

```diff
@@ -126,6 +142,29 @@ class LivyServer extends Logging {
     sys.props("livy.server.serverUrl") = _serverUrl.get
   }
 
+  def runKinit(livyConf: LivyConf): Runnable = new Runnable {
+    override def run(): Unit = {
+      UserGroupInformation.getCurrentUser.reloginFromTicketCache()
+      val keytab = livyConf.get(LAUNCH_KERBEROS_KEYTAB)
+      val principal = livyConf.get(LAUNCH_KERBEROS_PRINCIPAL)
+      require(principal != null,
+        s"Kerberos requires ${LAUNCH_KERBEROS_KEYTAB.key} to be provided.")
+      require(keytab != null,
+        s"Kerberos requires ${LAUNCH_KERBEROS_PRINCIPAL.key} to be provided.")
+      val commands = ArrayBuffer.empty[String]
+      commands += "kinit"
+      commands += "-kt"
+      commands += keytab
+      commands += principal
```

Comment:
Shouldn't you also pass `-l` so that you at least request a ticket with a lifetime long enough for the re-login period set up in the configuration? (I think it's `-l`, don't have the man page handy.)